### ISSUE TITLE
why did the model of SCL_PFENET trained by myself fail to achievel the performance of the paper? miou just 60.283

### ISSUE THREAD

**Issue by TizzyTTT** [zbf1991/SCL#4]
I  download the paper code and trained the model (SCL PFENET ) for 200 epochs without any change, pytorch version is 1.6.0.  But the best Miou is just 60.28

Long before , I also trained PFENet by myself. And I got the similar results mIoU 61.4~61.6

**Comment by TizzyTTT** [comment]
backbone: resnet50 
dataset: pasical 5i  fold0

**Comment by Saralyliu** [comment]
we kept with pytorch vision as PFENET, and got a similar result 61.4 as you.

**Comment by Saralyliu** [comment]
we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.

**Comment by TizzyTTT** [comment]
> 
> 
> we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
> The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.

I think the framework of PEFNet has no problems. I' ve also tried other splits and there is not much difference with the paper.  How about changing the seed number or make the batch size 6 ? or change a GPU.

**Comment by TizzyTTT** [comment]
> 
> 
> we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
> The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.
How about updating the version of pytorch ? A higher version pytorch maybe plays a role.

**Comment by TizzyTTT** [comment]
> 
> 
> > we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
> > The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.
> 
> I think the framework of PEFNet has no problems. I' ve also tried other splits and there is not much difference with the paper. How about changing the seed number or make the batch size 6 ? or change a GPU.


How about updating the version of pytorch ? A higher version pytorch maybe plays a role.

**Comment by TizzyTTT** [comment]
> we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
> The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.

How about updating the version of pytorch ? A higher version maybe plays a role.

**Comment by Saralyliu** [comment]
> > we kept with pytorch vision as PFENET, trained the model SCL and only got a result 61.4.
> > The PFENET, we got 60.89, how your reproduced the PFENet, did you try other splits? I can't got the results.
> 
> I think the framework of PEFNet has no problems. I' ve also tried other splits and there is not much difference with the paper. How about changing the seed number or make the batch size 6 ? or change a GPU.

The author of PFENET suggested 1.4.0 pythorch and a GPu, we followed the setting, without change the code and cannot get the  
result. what are your setting details.

**Comment by zbf1991** [comment]
I will check again for the code to see if I have uploaded something incorrectly. Sorry for that

**Comment by zbf1991** [comment]
I am re-running my code and I will reply to you as soon as possible.

**Comment by zbf1991** [comment]
Hello, I have re-run my code, using one 2080Ti and the pytorch version is 1.4. before 50 epochs (46 epoch), it has reached 62.5% mIoU. I am also testing 1080Ti and pytorch 1.3.  If you can change "train.py" line 271, "epoch % 2" to "epoch%1" and re-run this code (I have not done this). I will keep this issue open and for any results please tell me.

**Comment by TizzyTTT** [comment]
> 
> 
> Hello, I have re-run my code, using one 2080Ti and the pytorch version is 1.4. before 50 epochs (46 epoch), it has reached 62.5% mIoU. I am also testing 1080Ti and pytorch 1.3. If you can change "train.py" line 271, "epoch % 2" to "epoch%1" and re-run this code (I have not done this). I will keep this issue open and for any results please tell me.


Thank you for replying. I' ll check it again.

**Comment by zbf1991** [comment]
I have re-uploaded my 1-shot file, which is the current file I just re-tun. Although there is no difference. You can download this project and re-run it, Besides, 1080Ti with pytorch1.4 is also okay (epoch 37 can reach 63.0). 2080ti with pytorch1.3 can also work. And it seems that pytorch1.3 is more stable. If you want to see the log file I can share it with you.

**Comment by TizzyTTT** [comment]
> 
> 
> I have re-uploaded my 1-shot file, which is the current file I just re-tun. Although there is no difference. You can download this project and re-run it, Besides, 1080Ti with pytorch1.4 is also okay (epoch 37 can reach 63.0). 2080ti with pytorch1.3 can also work. And it seems that pytorch1.3 is more stable. If you want to see the log file I can share it with you.

Feel  sorry,  I change the batch size from 4 to 6 last time. This time I keep all the setting as yours and the performance reach the 62.85 in epoch 46 (resnet50 pascal5i fold0  miou 0.6285).   it is very close

**Comment by TizzyTTT** [comment]
> 
> 
> I have re-uploaded my 1-shot file, which is the current file I just re-tun. Although there is no difference. You can download this project and re-run it, Besides, 1080Ti with pytorch1.4 is also okay (epoch 37 can reach 63.0). 2080ti with pytorch1.3 can also work. And it seems that pytorch1.3 is more stable. If you want to see the log file I can share it with you.

Feel sorry, I change the batch size from 4 to 6 last time. This time I keep all the setting as yours and the performance reaches the 62.85 in epoch 46 (resnet50 pascal5i fold0 miou 0.6285). it is very close

**Comment by zbf1991** [comment]
> > I have re-uploaded my 1-shot file, which is the current file I just re-tun. Although there is no difference. You can download this project and re-run it, Besides, 1080Ti with pytorch1.4 is also okay (epoch 37 can reach 63.0). 2080ti with pytorch1.3 can also work. And it seems that pytorch1.3 is more stable. If you want to see the log file I can share it with you.
> 
> Feel sorry, I change the batch size from 4 to 6 last time. This time I keep all the setting as yours and the performance reaches the 62.85 in epoch 46 (resnet50 pascal5i fold0 miou 0.6285). it is very close.

Thanks for your reply. Note that the val_pair in train.py is 2000, you can change it to 5000 to get the final performance, as done in the original PFENET. If there is no further problem, I will close this issue.
